### PR TITLE
Handling disconnected state better.

### LINKED ISSUE #68: If we are "Offline", "Errors" is also displayed.
We should clearly distinguish error state of offline state.

### LINKED ISSUE #70: When RPC is offline we should disable / debounce queries.
screenshot from 2016-04-27 21 50 08

It takes around 2seconds (of unresponsiveness) to handle all error actions coming from RPC.
When we detect that node is down, we should send single query (just to check if it's up) or debounce/group error actions.

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes #68 
Closes #70

### CODE REVIEW

File: client/provider/web3-provider.spec.js [hunk 1]
```diff
@@ -65,4 +65,16 @@ describe('WEB3 PROVIDER', () => {
...
+  it('should call onTickWhenDisconnected when you are disconnected', () => {
+    // given
+    cut.onTickWhenDisconnected = sinon.spy();
+    state.status.disconnected = true;
+
+    // when
+    cut.onTick();
+
+    // then
+    expect(cut.onTickWhenDisconnected.called).to.equal(true);
```
Comment (Adam-Goldman7) [inline client/provider/web3-provider.spec.js]: I would also check that rest of this.tickArr[...] weren't being called

Comment (tomusdrw) [inline client/provider/web3-provider.spec.js]: Fixed - but don't want to iterate through all methods.

File: client/provider/web3-provider.spec.js [hunk 2]
```diff
@@ -66,15 +67,17 @@ describe('WEB3 PROVIDER', () => {
...
 
     // when
     cut.onTick();
 
     // then
-    expect(cut.onTickWhenDisconnected.called).to.equal(true);
+    expect(web3.eth.getBlockNumber.called).to.equal(true);
+    expect(web3.eth.getHashrate.called).to.equal(false);
+    expect(web3.eth.getCoinbase.called).to.equal(false);
+    expect(web3.net.getPeerCount.called).to.equal(false);
```
Comment (Adam-Goldman7) [inline client/provider/web3-provider.spec.js]: 1. `.to.equal(false)` -> .to.be.false
2. maybe do a loop on all functions that run on tick?
3. not sure which label I should put here

Comment (tomusdrw) [inline client/provider/web3-provider.spec.js]: Fixed. (Altough I prefer simple assertions (even if it means a lot of duplication))

**Pull request merged**